issuetracker/src/main/java/org/openjdk/skara/issuetracker/jira/JiraProject.java

```diff
@@ -506,18 +506,20 @@ private String toTimeString(ZonedDateTime time) {
         return timeZoned.format(DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm"));
     }
 
-    private ArrayList<IssueTrackerIssue> queryIssues(String jql) {
-        var ret = new ArrayList<IssueTrackerIssue>();
+    private List<IssueTrackerIssue> queryIssues(String jql) {
+        var ret = new HashMap<String, IssueTrackerIssue>();
```

Comment:
I think we should try to preserve the order we received issues. We can do that if we use a `LinkedHashMap` here. It would also make more sense to me to use ascending order. Then we will include any concurrently updated issue in this call instead of waiting for next call. I also think it makes sense for the bot to process issues in ascending order, so the first updated issue is processed first.

Reply:
After offline discussion, I take this back, descending order is the correct solution. We could still preserve the order here, or maybe just sort the result on "updated" before returning it.

Reply:
Or just skip sorting.